src/runner.jl

```diff
@@ -151,10 +151,11 @@ function gen_runner_code(testfilename, logfilename, test_args)
         end
 
         # Flatten before calling `report` to avoid a `deepcopy`.
+        flattened_testsets = TestReports.flatten_results!(ts)
         open($(repr(logfilename)), "w") do io
-            prettyprint(io, report(TestReports.flatten_results!(ts)))
+            prettyprint(io, report(flattened_testsets))
         end
-        any_problems(ts) && exit(TestReports.TESTS_FAILED)
+        any_problems(flattened_testsets) && exit(TestReports.TESTS_FAILED)
```

Comment:
We're now properly testing against the scenario in #118 with the new "FailedNestedTest" package. By chaing this line to:
```julia
        any_problems(ts) && exit(TestReports.TESTS_FAILED)
```
results in the following test failure:
```
Test Summary:    | Pass  Fail  Total  Time
FailedNestedTest |    1     1      2  0.1s
  nested         |          1      1  0.1s
[ Info: FailedNestedTest tests passed. Results saved to /Users/cvogt/.julia/dev/TestReports/test/testlog.xml.
Test packages: Test Failed at /Users/cvogt/.julia/dev/TestReports/test/utils.jl:132
  Expression: TestReports.test(pkg)
    Expected: TestReports.PkgTestError
  No exception thrown
```